x-pack/plugins/upgrade_assistant/public/application/components/overview/deprecation_logging_toggle.tsx

```diff
@@ -38,73 +52,129 @@ const i18nTexts = {
+  const { data, error, isLoading, resendRequest } = api.useLoadDeprecationLogging();
 
-      setIsLoading(false);
+  const [isEnabled, setIsEnabled] = useState<boolean | undefined>(undefined);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [updateError, setUpdateError] = useState<ResponseError | undefined>(undefined);
 
-      if (responseError) {
-        setError(responseError);
-      } else if (data) {
-        setIsEnabled(data.isEnabled);
-      }
-    }
-
-    getDeprecationLoggingStatus();
-  }, [api]);
-
-  if (isLoading) {
-    return <EuiLoadingSpinner size="l" />;
-  }
-
-  const renderLoggingState = () => {
-    if (error) {
-      return i18nTexts.toggleErrorLabel;
+  useEffect(() => {
+    if (isLoading === false && data) {
+      setIsEnabled(data.isEnabled);
     }
-
-    return i18nTexts.toggleLabel;
-  };
+  }, [data, isLoading]);
 
   const toggleLogging = async () => {
     const newIsEnabledValue = !isEnabled;
 
-    setIsLoading(true);
+    setIsUpdating(true);
 
-    const { data, error: updateError } = await api.updateDeprecationLogging({
+    const {
+      data: updatedLoggingState,
+      error: updateDeprecationError,
+    } = await api.updateDeprecationLogging({
       isEnabled: newIsEnabledValue,
     });
 
-    setIsLoading(false);
+    setIsUpdating(false);
 
-    if (updateError) {
-      setError(updateError);
-    } else if (data) {
-      setIsEnabled(data.isEnabled);
+    if (updateDeprecationError) {
+      setUpdateError(updateDeprecationError);
+    } else if (updatedLoggingState) {
+      setIsEnabled(updatedLoggingState.isEnabled);
       notifications.toasts.addSuccess(
-        data.isEnabled ? i18nTexts.enabledMessage : i18nTexts.disabledMessage
+        updatedLoggingState.isEnabled ? i18nTexts.enabledMessage : i18nTexts.disabledMessage
       );
     }
   };
 
-  return (
-    <EuiSwitch
-      data-test-subj="upgradeAssistantDeprecationToggle"
-      label={renderLoggingState()}
-      checked={isEnabled}
-      onChange={toggleLogging}
-      disabled={isLoading || Boolean(error)}
-    />
-  );
+  if (isLoading || isUpdating) {
+    return (
+      <SectionLoading inline>
+        {isUpdating ? i18nTexts.updatingLabel : i18nTexts.loadingLabel}
+      </SectionLoading>
+    );
+  }
+
+  if (error) {
+    return (
+      <EuiCallOut
+        size="s"
+        title={i18nTexts.fetchErrorMessage}
+        iconType="alert"
+        color="danger"
+        data-test-subj="fetchLoggingError"
+      >
+        <EuiButton size="s" iconType="refresh" color="danger" onClick={resendRequest}>
+          {i18nTexts.reloadButtonLabel}
+        </EuiButton>
+      </EuiCallOut>
+    );
+  }
+
+  if (typeof isEnabled !== 'undefined') {
+    return (
+      <>
+        {updateError && (
+          <>
+            <EuiCallOut
+              data-test-subj="updateLoggingError"
+              size="s"
+              title={i18nTexts.updateErrorMessage}
```

Comment:
How about adding a description with some error details, like status code and message?